ports/zephyr/mpconfigport_full.h

```diff
@@ -0,0 +1,164 @@
+#define MICROPY_PY_MACHINE_SPI_LSB (SPI_TRANSFER_LSB)
+#define MICROPY_PY_MACHINE_PIN_MAKE_NEW mp_pin_make_new
+#define MICROPY_PY_MACHINE_UART     (1)
+#define MICROPY_PY_MACHINE_UART_INCLUDEFILE "ports/zephyr/machine_uart.c"
+#define MICROPY_PY_STRUCT           (1)
+#define MICROPY_PY_OS_UNAME         (1)
+#ifdef CONFIG_NETWORKING
+// If we have networking, we likely want errno comfort
+#define MICROPY_PY_ERRNO            (1)
+#define MICROPY_PY_SOCKET           (1)
+#endif
+#ifdef CONFIG_BT
+#define MICROPY_PY_BLUETOOTH        (1)
+#ifdef CONFIG_BT_CENTRAL
+#define MICROPY_PY_BLUETOOTH_ENABLE_CENTRAL_MODE (1)
+#endif
+#define MICROPY_PY_BLUETOOTH_ENABLE_GATT_CLIENT (0)
+#endif
+#define MICROPY_PY_BINASCII         (1)
+#define MICROPY_PY_HASHLIB          (1)
+#define MICROPY_PY_OS               (1)
+#define MICROPY_PY_TIME             (1)
+#define MICROPY_PY_TIME_TIME_TIME_NS (1)
+#define MICROPY_PY_TIME_INCLUDEFILE "ports/zephyr/modtime.c"
+#define MICROPY_PY_ZEPHYR           (1)
+#define MICROPY_PY_ZSENSOR          (1)
+#define MICROPY_PY_SYS_MODULES      (1)
+#define MICROPY_LONGINT_IMPL (MICROPY_LONGINT_IMPL_LONGLONG)
+#define MICROPY_FLOAT_IMPL (MICROPY_FLOAT_IMPL_FLOAT)
+#define MICROPY_PY_BUILTINS_COMPLEX (1)
+#define MICROPY_ENABLE_SCHEDULER    (1)
+#define MICROPY_VFS                 (1)
+#define MICROPY_READER_VFS          (MICROPY_VFS)
+
+// fatfs configuration used in ffconf.h
+#define MICROPY_FATFS_ENABLE_LFN       (1)
+#define MICROPY_FATFS_LFN_CODE_PAGE    437 /* 1=SFN/ANSI 437=LFN/U.S.(OEM) */
+#define MICROPY_FATFS_USE_LABEL        (1)
+#define MICROPY_FATFS_RPATH            (2)
+#define MICROPY_FATFS_NORTC            (1)
+
+#define MICROPY_COMP_CONST_FOLDING  (0)
+#define MICROPY_COMP_CONST          (1)
+#define MICROPY_COMP_DOUBLE_TUPLE_ASSIGN (1)
+
+#define MICROPY_PY_THREAD                   (1)
+#define MICROPY_PY_THREAD_GIL               (0)
+#define MICROPY_PY_THREAD_GIL_VM_DIVISOR    (0)
+
+extern void mp_hal_signal_event(void);
+#define MICROPY_SCHED_HOOK_SCHEDULED mp_hal_signal_event()
+
+#define MICROPY_PY_SYS_PLATFORM "zephyr"
+
+#ifdef CONFIG_BOARD
+#define MICROPY_HW_BOARD_NAME "zephyr-" CONFIG_BOARD
+#else
+#define MICROPY_HW_BOARD_NAME "zephyr-generic"
+#endif
+
+#ifdef CONFIG_SOC
+#define MICROPY_HW_MCU_NAME CONFIG_SOC
+#else
+#define MICROPY_HW_MCU_NAME "unknown-cpu"
+#endif
+
+typedef int mp_int_t; // must be pointer size
+typedef unsigned mp_uint_t; // must be pointer size
+typedef long mp_off_t;
+
+#define MP_STATE_PORT MP_STATE_VM
+
+// extra built in names to add to the global namespace
+#define MICROPY_PORT_BUILTINS \
+    { MP_ROM_QSTR(MP_QSTR_open), MP_ROM_PTR(&mp_builtin_open_obj) },
+
+#if MICROPY_PY_THREAD
+#define MICROPY_EVENT_POLL_HOOK \
+    do { \
+        extern void mp_handle_pending(bool); \
+        extern void mp_hal_main_sem_take(); \
+        mp_handle_pending(true); \
+        MP_THREAD_GIL_EXIT(); \
+        mp_hal_main_sem_take(); \
```

Comment:
I'm not sure what `mp_hal_main_sem_take()` is used for... it seems that this semaphore is never given so I'm not sure what it's supposed to do?

Reply:
I saw this on the ESP32 port and figured it would be needed here sooner rather than later, but forgot to remove it when submitting this PR. I'll clean this up and add it again when (if) required.